cocotb/decorators.py

```diff
@@ -89,13 +89,12 @@ def __init__(self, inst):
 
         if inspect.iscoroutine(inst):
             self._natively_awaitable = True
-            self._coro = inst.__await__()
         elif inspect.isgenerator(inst):
             self._natively_awaitable = False
-            self._coro = inst
```

Comment:
In principle we could do the reverse here and make this line
```suggestion
            self._coro = types._GeneratorWrapper(inst)
```
but we're not really supposed to rely on `_GeneratorWrapper`.

Either way, I don't think we need to worry about it.